chia/wallet/wallet_node.py

```diff
@@ -528,6 +529,26 @@ async def _messages_to_resend(self) -> List[Tuple[Message, Set[bytes32]]]:
 
         return messages
 
+    async def _update_active_tx_states(self) -> None:
+        """
+        Refresh our view of the transaction's state in the peer network
+        """
+        while not self._shut_down:
+            try:
+                state_changes = []
+                await asyncio.sleep(self.active_tx_update_seconds)
+                if self.wallet_state_manager is None:
+                    continue
+                for tx_id in self.wallet_state_manager.tx_store.get_active_tx_states().keys():
+                    new_state = await self.wallet_state_manager.tx_store._update_active_tx_state(tx_id)
+                    state_changes.append(new_state)
+                    # TODO: write to db
+                if any(state_changes):
+                    self.wallet_state_manager.state_changed("active_transactions")
+            except asyncio.CancelledError:
+                self.log.info("_update_active_tx_states cancelled, exiting.")
+                raise
+
```

Comment:
Do we need this periodic updating functionality? It strikes me as another moving part in an application that already has too many.  We could just shoehorn this transaction status information on the request for transaction records with the method `.to_json_dict_convenience` which does other similar things.

Reply:
> It strikes me as another moving part in an application that already has too many.

Agreed! However, each change should be evaluated for user/dev benefit.
Lots of lines added for little benefit could be a reason to reject a PR, but "It adds more code" is not enough.

We need something. Right now, there are cases where you submit a transaction, and essentially know nothing until it finally ends up confirmed.

> We could just shoehorn this transaction status information on the request for transaction records with the method .to_json_dict_convenience

Just adding the current status to `get_transaction_records` would work if we are okay with the GUI periodically polling the wallet for all unconfirmed transactions. 

Pros of your suggested approach:
* Less code than the code already implemented here
* No DB changes

Cons of your suggested approach:
* It will increase the length of an already synchronous call
* Can't notify GUI of changes

We might also be able to do something in-between!

Reply:
> "It adds more code" is not enough.

To be clear, the concern is complexity not just raw code lines.  Consistency of wallet state is already shaky at best and the addition of a periodic timer that _changes_ that state just strikes me at least as headache waiting to happen.  This is of course a vibe/opinion though, nothing concrete I'll admit.

> Just adding the current status to get_transaction_records would work if we are okay with the GUI periodically polling the wallet for all unconfirmed transactions.

Isn't this what it does now?  Is there a problem with this?  If there is a problem, then the suggested place to meet in the middle would be to dispatch the event but not to update the DB in this periodic timer.